test/evp_pkey_provided_test.c

```diff
@@ -1963,7 +1981,7 @@ static int test_fromdata_dsa_fips186_4(void)
                                           fromdata_params), 1))
         goto err;
 
-    while (dup_pk == NULL) {
+    for (;;) {
         ret = 0;
```

Comment:
I think @slontis noted previously that this could go outside the for loop, or be removed entirely as its assigned below